### PR TITLE
update the DCU compile

### LINKED ISSUE #5414: Daily DCU test failed since 20241102
### Describe the bug

Since 20241102, the DCU test is failed because of compile error. This error is introduced by commits between 0f44046 (tested at 20241102, and is failed) and 3984296 (tested at 20241101, and is passed).

The error is related to force/stress, which may be introduced by commit e276480937.

@dyzheng , could you have a check?

Image

The error message is:
```
/public/home/abacus/abacus-dcu/source/module_hamilt_pw/hamilt_pwdft/kernels/rocm/stress_op.hip.cu:294:5: error: use of undeclared identifier '__syncwarp'; did you mean '__sync_swap'?
    __syncwarp();
    ^~~~~~~~~~
    __sync_swap
/public/home/abacus/abacus-dcu/source/module_hamilt_pw/hamilt_pwdft/kernels/rocm/stress_op.hip.cu:294:5: note: '__sync_swap' declared here
/public/home/abacus/abacus-dcu/source/module_hamilt_pw/hamilt_pwdft/kernels/rocm/stress_op.hip.cu:294:16: error: too few arguments to function call, expected at least 1, have 0
    __syncwarp();
    ~~~~~~~~~~ ^
[ 70%] Building CXX object source/module_io/CMakeFiles/io_basic.dir/read_wfc_to_rho.cpp.o
2 errors generated when compiling for gfx906.
[ 70%] Building CXX object source/module_cell/CMakeFiles/cell.dir/read_pp_vwr.cpp.o
CMake Error at device_rocm_generated_stress_op.hip.cu.o.cmake:180 (message):
  Error generating file
  /public/home/abacus/abacus-dcu/build-dcu/source/CMakeFiles/device_rocm.dir/module_hamilt_pw/hamilt_pwdft/kernels/rocm/./device_rocm_generated_stress_op.hip.cu.o


make[2]: *** [source/CMakeFiles/device_rocm.dir/build.make:147: source/CMakeFiles/device_rocm.dir/module_hamilt_pw/hamilt_pwdft/kernels/rocm/device_rocm_generated_stress_op.hip.cu.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

### Expected behavior

_No response_

### To Reproduce

_No response_

### Environment

_No response_

### Additional Context

_No response_

### Task list for Issue attackers (only for developers)

- [ ] Verify the issue is not a duplicate.
- [ ] Describe the bug.
- [ ] Steps to reproduce.
- [ ] Expected behavior.
- [ ] Error message.
- [ ] Environment details.
- [ ] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

### PR DESCRIPTION
**Pull request by A-006**
### Linked Issue
Fix #5414 

### What's changed?
- Change the DCU FFT. 
- With the help of @Qianruipku ,the complie link error has been finished.
- In the DCU file, i use the complie instruction as bellow
#!/bin/bash

PACKAGES=/work/home/acfvnd0rku/liu/packages
        CC=clang CXX=clang++ 
        cmake -B build -DUSE_OPENMP=OFF -DENABLE_LCAO=OFF 
        -DFFTW3_DIR=${PACKAGES}/fftw-3.3.9/build/ 
        -DLAPACK_DIR=${PACKAGES}/OpenBLAS-0.3.21/build/lib 
        -DSCALAPACK_DIR=${PACKAGES}/scalapack-2.2.0/ 
        -DUSE_ROCM=ON
cmake --build build -j50

and with the sbatch file as bellow

#!/bin/bash
#SBATCH --job-name=ABACUS_GPU 
#SBATCH --partition=work
#SBATCH --nodes=1 
#SBATCH --output=output.log    
#SBATCH --ntasks-per-node=32  
#SBATCH --mail-user=username@email  
#SBATCH --gres=dcu:4  #dcu个数
#SBATCH --time=01:00:00  
#SBATCH --error=error.log   

abacus=/work/home/acfvnd0rku/liu/fft2/abacus-develop/build/abacus_pw

module purge
module load compiler/intel
module load compiler/devtoolset/7.3.1
module load compiler/dtk/23.10
module load compiler/cmake/3.23.3
module load mpi/intelmpi/2021.3.0

#cd your_task_path 
mpirun -np 4 $abacus

the result is as follow:
img_v3_02gr_6e3c89ac-ca18-436b-a5d0-c0f1a1cccc6g

### CODE REVIEW

File: source/module_basis/module_pw/module_fft/fft_rcom.h [hunk 1]
```diff
@@ -6,11 +6,11 @@
 namespace ModulePW
 {
 template <typename FPTYPE>
-class FFT_ROCM : public FFT_BASE<FPTYPE>
+class FFT_RCOM : public FFT_BASE<FPTYPE>
```
Comment (caic99) [inline source/module_basis/module_pw/module_fft/fft_rcom.h]: Should be rocm right 🤣

Comment (A-006) [reply source/module_basis/module_pw/module_fft/fft_rcom.h]: So sorry for this problem.

Comment (caic99) [reply source/module_basis/module_pw/module_fft/fft_rcom.h]: Please also fix other typos in this PR.

Comment (A-006) [reply source/module_basis/module_pw/module_fft/fft_rcom.h]: Okay,thanks,it's done. Here is the current commit's DCU test status.Here are the test results using DCU for the current commit.
img_v3_02gt_61828b1c-ef16-46eb-a224-8b0144a92acg

Comment (A-006) [reply source/module_basis/module_pw/module_fft/fft_rcom.h]: This is the result from the single version.
img_v3_02gt_092d2ed0-5663-4ef3-9bc1-38d92c7d8feg

Comment (caic99) [reply source/module_basis/module_pw/module_fft/fft_rcom.h]: I'm sorry but I mean `rcom` in your PR should be `rocm`, right? Please correct me if I'm wrong.

> Here is the current commit's DCU test status.Here are the test results using DCU for the current commit.

Sorry but what's the point here?

Comment (A-006) [reply source/module_basis/module_pw/module_fft/fft_rcom.h]: Thank you for the reminder. I misunderstood earlier. The code has now been modified to rocm.

**Review by Qianruipku — APPROVED** [review]

**Pull request merged**